staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/azure_fileclient.go

```diff
@@ -45,14 +47,22 @@ func New(config *azclients.ClientConfig) *Client {
 
 // CreateFileShare creates a file share
 func (c *Client) CreateFileShare(resourceGroupName, accountName, name string, sizeGiB int) error {
+	result, err := c.GetFileShare(resourceGroupName, accountName, name)
+	if err == nil {
+		klog.V(2).Infof("file share(%s) under account(%s) rg(%s) already exists", name, accountName, resourceGroupName)
+		return nil
+	} else if err != nil && result.Response.StatusCode != http.StatusNotFound && !strings.Contains(err.Error(), "ShareNotFound") {
```

Comment:
could you add an unit test for this?

Reply:
Currently, could not add the unit test. Because `storage.FileSharesClient` is a struct, we can't mock it now. After updating the azure-sdk-for-go in the future, it will be switched to use interface. Then we can mock it.

Reply:
makes sense.